src/vos/vos_gc.c

```diff
@@ -296,6 +296,15 @@ gc_drain_cont(struct vos_gc *gc, struct vos_pool *pool, daos_handle_t coh,
 	int			 i;
 	int			 rc;
 
+	/*
+	 * When we prepaer to drain the container, we do not need DTX entry any long.
+	 * Then destroy DTX table firstly to avoid dangling DXT records during drain
+	 * the container (that may yield).
+	 */
+	rc = vos_dtx_table_destroy(&pool->vp_umm, cont);
```

Comment:
I don't see why it's moved here, then it'll called on draining every object? The original place looks more proper to me.

Reply:
Because in gc_drain_cont(), there is assertion:
`        D_ASSERT(daos_handle_is_inval(coh));
        return gc_drain_btr(gc, pool, coh, item, &cont->cd_obj_root, credits, empty);
`
Means that the `coh` parameter for dtx_deregister will always be zero for draining the container. That will cause related DTX entry to be left as dangling. To avoid such case, since it will be drained, the DTX table is useless any longer, we can destroyed the DTX table firstly. On the other hand, even if the `coh` is not zero, deregistering DTX records one by one is also inefficient for drain case, destroying the table will be more efficient.